docs/deploy.md

```diff
@@ -0,0 +1,66 @@
+# Deploying Cluster API Provider Openshift BareMetal
+
+
+## Deploy on vanilla Kubernetes
+This provider is configured to deploy on [Red Hat OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift) (OCP) by default.
+
+To deploy this provider on vanilla Kubernetes, the following must be done:
+
+### Prerequisites
+
+A vanilla Kubernetes such as [Kind](https://kind.sigs.k8s.io/) must be deployed and you must be authenticated to the cluster.
+
+The following services are required on your cluster before installing this provider.
+
+1. Install [Assisted-Service operator](https://github.com/openshift/assisted-service/blob/master/docs/dev/operator-on-kind.md)
+2. Install [CAPI](https://cluster-api.sigs.k8s.io/user/quick-start.html)
```

Comment:
We already have some instructions in the readme about installing the providers, can we integrate those docs with these new instructions?

Reply:
Gotcha, yes tried to do so but since we're suggesting a kustomize with a patch, these instructions/prerequisites mostly stayed the same 😅 do you know how to do it with clusterctl? 


Also would you like assisted service as a prerequisite for the general deployment instructions? I was assuming we're targeting OCP so assisted service would already be deployed and need not be listed as a general prerequisite. Only listed if it's vanilla kube. Let me know your thoughts!

Reply:
AFAIK we cannot do this with clusterctl, but I meant maybe we can have one section explaining how to install, with all the variants.